torch/lib/THD/base/channels/DataChannelTCP.cpp

```diff
@@ -276,18 +279,16 @@ int DataChannelTCP::connect(const std::string& address, std::uint16_t port,
       // if `connect` fails, the state of the socket is unspecified.
       // we should close the socket and create a new one before attempting to reconnect.
       ::close(socket);
+      if (errno == ECONNREFUSED) any_refused = true;
 
-      if (!wait || (errno != ECONNREFUSED)) {
-        // we need to move to next address because this was not available
-        // to connect or to create socket
-        next_addr = next_addr->ai_next;
+      // we need to move to next address because this was not available
+      // to connect or to create socket
+      next_addr = next_addr->ai_next;
 
-        // we have tried all addresses but could not connect to any of them
-        if (!next_addr) {
-          throw e;
-        }
-      } else {
-        std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
Not sure if removing this completely is good idea. If master is not yet initialized worker will spin in this loop and cause 100% cpu usage. It is better to change to like 100ms if we want to avoid long pauses.